enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/storecopy/StreamToDisk.java

```diff
@@ -63,7 +64,11 @@ public void write( String destination, int requiredAlignment, byte[] data ) thro
         fs.mkdirs( fileName.getParentFile() );
 
         fileCopyMonitor.copyFile( fileName );
-        if ( StoreType.shouldBeManagedByPageCache( destination ) )
+
+        // Default filesystem supports writing of data through the file system directly
+        boolean isDefaultFileSystem = pageCache.getCachedFileSystem() instanceof DefaultFileSystemAbstraction;
+
+        if ( !isDefaultFileSystem && StoreType.shouldBeManagedByPageCache( destination ) )
```

Comment:
same here with reversing condition

Reply:
I thik that inverted pattern with ` || !StoreType.shouldBeManagedByPageCache` is harder to parse, what do you think?